include/mbedtls/ssl.h

```diff
@@ -1505,15 +1503,45 @@ void mbedtls_ssl_conf_export_keys_cb( mbedtls_ssl_config *conf,
  * \param f_async_cancel    Callback to cancel an asynchronous operation. See
  *                          the description of \c mbedtls_ssl_async_cancel_t
  *                          for more information.
- * \param connection_ctx    Pointer to the connection context which will be
- *                          passed to the callbacks
+ * \param config_data       A pointer to configuration data which will be
+ *                          passed to the callbacks. The library stores and
+ *                          passes back this value without dereferencing it.
  */
 void mbedtls_ssl_conf_async_private_cb( mbedtls_ssl_config *conf,
                                         mbedtls_ssl_async_sign_t *f_async_sign,
                                         mbedtls_ssl_async_decrypt_t *f_async_decrypt,
                                         mbedtls_ssl_async_resume_t *f_async_resume,
                                         mbedtls_ssl_async_cancel_t *f_async_cancel,
-                                        void *connection_ctx );
+                                        void *config_data );
+
+/**
+ * \brief           Retrieve the asynchronous operation user context.
+ *
+ * \note            This function may only be called while a handshake
+ *                  is in progress.
+ *
+ * \param ssl       The SSL context to access.
+ *
+ * \return          The asynchronous operation user context that was last
+ *                  set during the current handshake. If mbedtls_ssl_set_data()
+ *                  has not been called during the current handshake yet,
+ *                  this function returns \c NULL.
+ */
+void *mbedtls_ssl_async_get_data( mbedtls_ssl_context *ssl );
```

Comment:
Looks like we could, hence should, make the parameter `const`.

Reply:
Done in a later commit.